### PR TITLE
Prometheus Plugin [WIP]

### PR DESCRIPTION
**Pull request by malnick**

### CODE REVIEW

**Comment by ksexton** [comment]
Any updates on this?  Would like to test out DC/OS 1.9 metrics with Prometheus.

**Comment by tobilg** [comment]
@collinvandyck When will this be merged?

**Comment by philipnrmn** [comment]
@tobilg there's a JIRA ticket for this here:
https://jira.mesosphere.com/browse/DCOS_OSS-894

**Comment by bawgeneric** [comment]
Great

**Comment by Chewie** [comment]
Any news regarding this?
In case nobody's working on this, is there interest in someone taking back ownership and clean it up?

File: plugins/prometheus/prometheus.go [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+func main() {
+	proFlags := []cli.Flag{
+		cli.StringFlag{
+			Name:  "port",
+			Value: ":8010",
+			Usage: "Port for exposing metrics to Prometheus",
+		},
+	}
+
+	proPlugin, err := plugin.New(proFlags)
```
Comment (eolix) [inline plugins/prometheus/prometheus.go]: This spits a compilation-time error:

> plugins/prometheus/prometheus.go:42: cannot use proFlags (type []"github.com/dcos/dcos-metrics/vendor/github.com/urfave/cli".Flag) as type []"github.com/malnick/dcos-metrics-plugins/vendor/github.com/urfave/cli".Flag in argument to plugin.New

File: plugins/prometheus/prometheus.go [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+	proPlugin.App.Action = func(c *cli.Context) {
+		metrics, err := proPlugin.Metrics()
+		if err != nil {
+			logrus.Fatal(err)
+		}
+
+		go func() {
+			for {
+				err = registerMetrics(metrics)
```
Comment (eolix) [inline plugins/prometheus/prometheus.go]: This spits a compilation-time error:
> plugins/prometheus/prometheus.go:55: cannot use metrics (type []"github.com/malnick/dcos-metrics-plugins/vendor/github.com/dcos/dcos-metrics/producers".MetricsMessage) as type []"github.com/dcos/dcos-metrics/producers".MetricsMessage in argument to registerMetrics

**Comment by philipnrmn** [comment]
@eolix You have collisions between the vendored cli package in the `dcos-metrics-plugins` repository belonging to @malnick and our vendored cli package. I'm working to untangle this :)

**Comment by philipnrmn** [comment]
Closing this in favour of #127

**Pull request closed without merging**